### ISSUE TITLE
Tokenizer训练相关问题

### ISSUE THREAD

**Issue by 1059915154** [shiyu-coder/Kronos#200]
首先感谢作者的分享。
  请教一下微调Tokenizer时碰到一些问题。训练数据为ETH分钟级K线，大概300w多。但是微调后的效果与原模型差距很大。
以下是相关参数：
  clip: 5.0
  train_ratio: 0.9
  val_ratio: 0.1
  batch_size: 50
  tokenizer_learning_rate: 0.0002
  adam_beta1: 0.9
  adam_beta2: 0.95
  adam_weight_decay: 0.1
部分训练日志：
2025-11-30 17:06:23 - tokenizer_training_rank_0 - INFO -   - VQ Loss: -0.0714
  - Recon Loss Pre: 0.0076
  - Recon Loss All: 0.0042
  - VQ Loss: -0.0714
  - Recon Loss Pre: 0.0068
  - Recon Loss All: 0.0037
[Epoch 3/30, Step 62220/66390] LR: 0.000198, Loss: -0.0306
  - VQ Loss: -0.0715
  - Recon Loss Pre: 0.0067
  - Recon Loss All: 0.0036
[Epoch 3/30, Step 63220/66390] LR: 0.000198, Loss: -0.0299
  - VQ Loss: -0.0714
  - Recon Loss Pre: 0.0076
  - Recon Loss All: 0.0041
[Epoch 3/30, Step 64220/66390] LR: 0.000198, Loss: -0.0301
  - VQ Loss: -0.0713
  - Recon Loss Pre: 0.0070
  - Recon Loss All: 0.0042
[Epoch 3/30, Step 65220/66390] LR: 0.000197, Loss: -0.0311
  - VQ Loss: -0.0716
  - Recon Loss Pre: 0.0059
  - Recon Loss All: 0.0035
[Epoch 3/30, Step 66220/66390] LR: 0.000197, Loss: -0.0298
  - VQ Loss: -0.0714
  - Recon Loss Pre: 0.0076
  - Recon Loss All: 0.0042
2025-11-30 17:09:12 - tokenizer_training_rank_0 - INFO - 
--- Epoch 3/30 Summary ---
Validation Loss: 0.0027
Epoch Time: 1:15:58
Total Training Time: 1:15:58
2025-11-30 17:09:12 - tokenizer_training_rank_0 - INFO - Best model saved to: ../outputs/models/eth_base/tokenizer/best_model (validation loss: 0.0027)

以下是基于微调的Tokenizer预测结果：
<img width="1148" height="632" alt="Image" src="https://github.com/user-attachments/assets/911c6be0-c8cc-43e4-9ea7-8c6afe1975de" />
以下是基于原Tokenizer预测结果：
<img width="1359" height="753" alt="Image" src="https://github.com/user-attachments/assets/4f63c335-9b13-41ba-9486-ea5a094cecef" />

而且基于微调后的Tokenizer进行微调kronos-base时，v_loss很难降到3以下。但是基于原Tokenizer微调kronos-base时，v_loss在2.2-2.4之间。
顺便再请教下微调kronos-base问题，很容易过拟合，以至于我实际lr已经降到1e-6，虽然v_loss有略微降低，但训练loss其实没下降趋势。建议扩大训练集（添加其他市场）还是增强正则化？亦或者有其他更好的建议？

**Comment by shiyu-coder** [comment]
微调tokenizer后，kronos需要使用微调后的tokenizer进行微调（就是继续新的tokenizer再对kronos进行微调），不然码表不再是对应的了。时序基础模型的微调确实很容易越调越差，这一点在其他paper中也被观察到了。我的经验就是小的学习率多训练一会，尽可能数据量大一些，但是也不是一直都有效。

**Comment by shawn-zxw** [comment]
@shiyu-coder 感谢回复，再请教一下相关参数设置。
1.adam_weight_decay、batch_size在训练tokenizer和kronos需要保持一致吗？分别推荐设置多少合适？
2.tokenizer_learning_rate：有看到你在其他issues提到tokenizer_learning_rate设置可以大一点，2e-4合适吗？

**Comment by world2025** [comment]
@shiyu-coder 请问下微调base_model的时候，必须要先训练好tokenizer吗，会指定加载tokenizer的best_model

<img width="652" height="66" alt="Image" src="https://github.com/user-attachments/assets/dd360959-586a-49a7-b415-6a64a2520bb5" />

**Comment by shiyu-coder** [comment]
可以跳过tokenizer的微调，直接对模型进行微调的，tokenizer的微调不是必须的

**Comment by world2025** [comment]
> 可以跳过tokenizer的微调，直接对模型进行微调的，tokenizer的微调不是必须的

我知道，但是我用finetune_csv的finetune_base_model.py脚本的时候，不是加载预训练模型，而是默认加载best_model

**Comment by shiyu-coder** [comment]
这个自己修改一下就可以了，改成直接加载预训练模型

> > 可以跳过tokenizer的微调，直接对模型进行微调的，tokenizer的微调不是必须的
> 
> 我知道，但是我用finetune_csv的finetune_base_model.py脚本的时候，不是加载预训练模型，而是默认加载best_model

**Comment by world2025** [comment]
> 这个自己修改一下就可以了，改成直接加载预训练模型
> 
> > > 可以跳过tokenizer的微调，直接对模型进行微调的，tokenizer的微调不是必须的
> > 
> > 
> > 我知道，但是我用finetune_csv的finetune_base_model.py脚本的时候，不是加载预训练模型，而是默认加载best_model
好的，我修改了下代码，不然就是一直默认用finetune好的tokenizer。然后就是请问下，训练数据的时间戳格式是2019/11/26 9:35:00，我自己的数据是只有到天，如果要适配自己的数据，是需要修改时间戳这部分代码是吗
